### ISSUE TITLE
Node standard library module "node:crypto".

### ISSUE THREAD

**Issue by caglar09** [replicate/replicate-javascript#210]
When I install version 0.27.1 of replicate-javascript library in react-native project, I get error like below. 

Node Version: 18.18.0

`The package at "node_modules/replicate/lib/util.js" attempted to import the Node standard library module "node:crypto".
It failed because the native React runtime does not include the Node standard library`

When I install version 0.26.0, it does not give any error. The problem seems to be related to the improvements made in the newly released version 0.27.1. 

**How can we find a solution for this?**

---
Update: 

I don't use webhooks in the project. I applied a temporary patch like below, but I haven't tested it for webhooks!

```
diff --git a/node_modules/replicate/lib/util.js b/node_modules/replicate/lib/util.js
index 48d7563..6ffc9bd 100644
--- a/node_modules/replicate/lib/util.js
+++ b/node_modules/replicate/lib/util.js
@@ -1,4 +1,4 @@
-const crypto = require("node:crypto");
+const CryptoJS = require("crypto-js");
 
 const ApiError = require("./error");
 
@@ -74,10 +74,7 @@ async function validateWebhook(requestData, secret) {
 
   const secretBytes = Buffer.from(signingSecret.split("_")[1], "base64");
 
-  const computedSignature = crypto
-    .createHmac("sha256", secretBytes)
-    .update(signedContent)
-    .digest("base64");
+  const computedSignature = CryptoJS.HmacSHA256(secretBytes, signedContent).toString(CryptoJS.enc.Base64);
 
   const expectedSignatures = signature
     .split(" ")
diff --git a/node_modules/replicate/package.json b/node_modules/replicate/package.json
index 098bb3a..3ff64b6 100644
--- a/node_modules/replicate/package.json
+++ b/node_modules/replicate/package.json
@@ -33,7 +33,8 @@
     "test": "jest"
   },
   "optionalDependencies": {
-    "readable-stream": ">=4.0.0"
+    "readable-stream": ">=4.0.0",
+    "crypto-js":"4.2.0"
   },
   "devDependencies": {
     "@biomejs/biome": "^1.4.1",

```

**Comment by aron** [comment]
Hey @caglar09, I've opened up a patch to fix this in #216, if you have the time and are able to test it in a React Native environment I'd appreciate it!

**Comment by guivr** [comment]
node:crypto is still not available in React Native

![IMG_2848](https://github.com/replicate/replicate-javascript/assets/7696343/6bba2cb4-8634-49a5-a4ff-f6e3c91cc3e9)

**Comment by aron** [comment]
@guivr would you please provide some additional details about the setup of your project or a simple example project that demonstrates the issue? Currently I'm able to build and run a React native project that has a dependency on Replicate and run it.

I've attempted to get a web application running using the [React Native tutorial](https://reactnative.dev/docs/environment-setup) and am able to import and run the replicate library fine and launch the web version of the app. So understanding what versions of react-native you're using as well as the platforms and build setup would be helpful to understanding the problem further.

**Comment by guivr** [comment]
@aron yes I've also used that tutorial and created the app via the `npx create-expo-app AwesomeProject` command.

The platform is iOS. I'm running on my own iPhone (13 Pro).

**Comment by mattt** [comment]
Hi @caglar09 & @guivr. We just released [v0.29.3](https://github.com/replicate/replicate-javascript/releases/tag/v0.29.3), which includes changes in #250 that _should_ address this. Please give that a whirl and let us know if things still aren't working for you 🙏
